cmd/trace-agent/config/config.go

```diff
@@ -383,6 +383,10 @@ func applyDatadogConfig(c *config.AgentConfig) error {
 		c.DDAgentBin = coreconfig.Datadog.GetString("apm_config.dd_agent_bin")
 	}
 
+	if coreconfig.Datadog.IsSet("apm_config.go_memlimit_pct") {
```

Comment:
Let's also bind the corresponding env here https://github.com/DataDog/datadog-agent/blob/main/pkg/config/apm.go (`config.BindEnv("apm_config.go_memlimit_pct", "DD_APM_GO_MEMLIMIT_PCT")`